source/ncasset/BonesReader.cpp

```diff
@@ -0,0 +1,43 @@
+#include "BonesReader.h"
+
+namespace nc::asset
+{
+auto ReadMatrix(RawNcaBuffer& bytes) -> DirectX::XMMATRIX
+{
+    float buf[16];
+    bytes.Read(buf, 16 * sizeof(float));
+    return DirectX::XMMATRIX{buf};
+}
+
+void Read(RawNcaBuffer& bytes, std::vector<nc::asset::VertexSpaceToBoneSpace>* vertexSpaceToBoneSpaceMatrices, size_t matrixCount)
+{
+    for (auto i = 0u; i < matrixCount; i++)
+    {
+        auto boneNameSize = size_t{};
+        bytes.Read(&boneNameSize);
+        auto vertexSpaceToBoneSpace = nc::asset::VertexSpaceToBoneSpace{};
+        bytes.Read(vertexSpaceToBoneSpace.boneName.data(), boneNameSize);
```

Comment:
Any dynamically-sized data should be resized appropriately before writing into it. This case looks like a write into an empty string. There's another instance like this in the other read function.

I think you have test coverage of this, so I wonder why it isn't causing issues? Maybe try testing with an super long string and see if it kabooms.

Reply:
Big bada boom with a big string. Good catch!